components/abstractions/src/main/java/com/microsoft/kiota/store/InMemoryBackingStore.java

```diff
@@ -112,20 +126,13 @@ public void clear() {
         return result;
     }
 
-    private Object getValueFromWrapper(final String entryKey, final Pair<Boolean, Object> wrapper) {
+    private Object getValueFromWrapper(final Pair<Boolean, Object> wrapper) {
         if (wrapper != null) {
-            final Boolean hasChanged = wrapper.getValue0();
-            if (!this.returnOnlyChangedValues || Boolean.TRUE.equals(hasChanged)) {
-                if (Boolean.FALSE.equals(
-                        hasChanged)) { // no need property has already been flagged.
-                    ensureCollectionPropertyIsConsistent(entryKey, wrapper.getValue1());
-                }
-                if (wrapper.getValue1() instanceof Pair) {
-                    Pair<?, ?> collectionTuple = (Pair<?, ?>) wrapper.getValue1();
-                    return collectionTuple.getValue0();
-                }
-                return wrapper.getValue1();
+            if (wrapper.getValue1() instanceof Pair) {
```

Comment:
We could do an early return, like a null check followed by an early return, will be more readable and reduce nesting

```java
 if (wrapper == null) {
        return null;
    }
```

Reply:
And maybe a javadoc comment